core/primitives/src/stateless_validation.rs

```diff
@@ -17,10 +18,30 @@ use near_primitives_core::types::{AccountId, Balance, BlockHeight, ShardId};
 /// This is a messy workaround until we know what to do with NEP 483.
 type SignatureDifferentiator = String;
 
-/// Signable
 #[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
-pub struct ChunkStateWitness {
-    pub inner: ChunkStateWitnessInner,
+pub struct EncodedChunkStateWitness(Box<[u8]>);
```

Comment:
can you add comments to this one and below (signed witness) to briefly explain what they represent and how they are used? thanks.

Reply:
sure, done in e1a843fea0f1db1f538da57b66030d346661843c